lib/collection/certificate.js

```diff
@@ -0,0 +1,129 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+    UrlMatchPatternList = require('../url-pattern/url-match-pattern-list').UrlMatchPatternList,
+
+    Certificate;
+
+/**
+* The following is the object representation accepted as param for the Certificate constructor.
+* Also the type of the object returned when {@link Property#toJSON} or {@link Property#toObjectResolved} is called on a
+* Certificate instance.
+* @typedef Certificate~definition
+* @property {String} [name] A name for the certificate
+* @property {Array} [matches] A list of match patterns
+* @property {String} [keyPath] The path on the file system to the file that contains the key
+* @property {String} [pemPath] The path on the file system to the certificate file
+* @property {String} [passphrase] The passphrase for the certificate key
+*
+* @example <caption>JSON definition of an example certificate object</caption>
+* {
+*     name: 'My certificate for example.com',
```

Comment:
We analysed possible types of certificates we might want to support. 

Broadly the certificates can be Private Key, and Certificate in a same file (`pfx`) or in individual files (`pem`). So the data structure would look like

```js
{
  key: ...,
  cert: ...
}
```
and 
```
{
    pfx: ...
}
```

I deferred adding support to `pfx` now because `postman-runtime` does not support it now, and it can extended without breaking the current structure.